src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/CoreCLR/WebRequestPSCmdlet.CoreClr.cs

```diff
@@ -347,14 +391,24 @@ protected override void ProcessRecord()
                         WriteVerbose(reqVerboseMsg);
 
                         HttpResponseMessage response = GetResponse(client, request);
-                        response.EnsureSuccessStatusCode();
 
                         string contentType = ContentHelper.GetContentType(response);
                         string respVerboseMsg = string.Format(CultureInfo.CurrentCulture,
                             "received {0}-byte response of content type {1}",
                             response.Content.Headers.ContentLength,
                             contentType);
                         WriteVerbose(respVerboseMsg);
+
+                        if (!response.IsSuccessStatusCode)
+                        {
+                            string message = String.Format(CultureInfo.CurrentCulture, WebCmdletStrings.ResponseStatusCodeFailure,
+                                response.StatusCode.ToString(), response.ReasonPhrase);
+                            HttpResponseException httpEx = new HttpResponseException(message);
```

Comment:
You could use this syntax here for creating the HttpResponseException:
```
var httpEx = new HttpResponseException(message)
{
    Response = response,
    Status = ...
    Headers = ...
};
```
Or you could make the constructor take all property values and make the properties read only.